code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -149,17 +149,18 @@
 			user.visible_message("<span class='danger'>The shotgun goes off!</span>", "<span class='danger'>The shotgun goes off in your face!</span>")
 			return
 		if(do_after(user, src, 30))	//SHIT IS STEALTHY EYYYYY
-			icon_state = "sawnshotgun"
-			w_class = W_CLASS_MEDIUM
-			item_state = "sawnshotgun"
-			slot_flags &= ~SLOT_BACK	//you can't sling it on your back
-			slot_flags |= SLOT_BELT		//but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
-			name = "sawn-off shotgun"
-			desc = "Omar's coming!"
+			var/obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff/itssmallnow = new /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff(src.loc)
+			for(var/obj/item/ammo_casing/shotgun/loaded_shell in itssmallnow)
```

Comment:
you could make a subtype that starts with no shells in the first place instead

Reply:
i've had bad experiences with such things in the past but that'd probably work
is it necessary, though?

Reply:
It's not a big deal either way, but I don't like creating things (the bullets in the shotgun's New) only to delete them immediately after. It's not free, but it's also not expensive.

Reply:
makes sense